lib/TH/generic/THTensorMath.c

```diff
@@ -99,7 +99,7 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
   long dim;
   long div = 1;
 #ifdef TH_REAL_IS_HALF
-#define IS_NONZERO(val) (TH_half2float(val)!=0)
```

Comment:
that's fine, I was just suggesting you either use TH_HALF_ZERO or get rid of it.